packages/flutter/lib/src/material/material_app.dart

```diff
@@ -141,8 +141,9 @@ class _MaterialAppState extends State<MaterialApp> implements BindingObserver {
       data: new MediaQueryData(size: _size),
       child: new LocaleQuery(
         data: _localeData,
-        child: new Theme(
+        child: new AnimatedTheme(
```

Comment:
I'll check a release mode build. It wasn't horrific in checked mode. I was a bit skeptical too, but honestly before it was doing a whole bunch of rebuilding anyway.